app/services/process_mentions_service.rb

```diff
@@ -41,6 +42,12 @@ def call(status)
       "@#{mentioned_account.acct}"
     end
 
+    if circle.present?
+      circle.accounts.find_each do |target_account|
+        status.mentions.create(silent: true, account: target_account)
+      end
+    end
```

Comment:
Won't this fail if someone from the circle is also explicitly mentioned? Because of the uniqueness constraint to mentions.

Reply:
Because it is not added to the mentions, it will not be created_notification().

```
mention = target_account.mentions.new(status: status, silent: true)
mentions << mention if mention.save
```

Reply:
True. In addition, `create_notification` is kind of a misnomer because it handles both notifications and AP federation. In this case we probably do not want notifications (unless explicit mention), but we want to federate through AP.